src/Integration/LocalServices/RuleSetInspector.cs

```diff
@@ -0,0 +1,144 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="RuleSetInspector.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.CodeAnalysis.Extensibility;
+using Microsoft.VisualStudio.CodeAnalysis.RuleSets;
+using Microsoft.VisualStudio.Shell.Interop;
+using SonarLint.VisualStudio.Integration.Resources;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+
+namespace SonarLint.VisualStudio.Integration
+{
+    internal class RuleSetInspector : IRuleSetInspector
+    {
+        private readonly IServiceProvider serviceProvider;
+        private readonly HashSet<string> ruleSetSearchDirectories = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// Relative path from VS install to the rule sets folder that come with VS
+        /// </summary>
+        public const string DefaultVSRuleSetsFolder = @"Team Tools\Static Analysis Tools\Rule Sets";
+
+        public RuleSetInspector(IServiceProvider serviceProvider, params string[] userRuleSetDirectories)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            this.serviceProvider = serviceProvider;
+            ruleSetSearchDirectories.UnionWith(userRuleSetDirectories);
+            ruleSetSearchDirectories.Add(GetStaticAnalysisToolsDirectory());
+        }
+
+        public RuleConflictInfo FindConflictingRules(string baseLineRuleSet, string targetRuleSet)
```

Comment:
Probably worth starting reviewing from here. It provides all the information you should need to understand how rules are merged and the kind of things we need to make sure we support in general. The information was gather by looking at the implementation of RuleSet.GetEffectiveRules.